module-apps/application-messages/windows/SMSTemplatesWindow.cpp

```diff
@@ -128,4 +131,18 @@ namespace gui
             application,
             std::make_unique<app::manager::SwitchBackRequest>(nameOfPreviousApplication.value(), nullptr, true));
     }
+
+    bool SMSTemplatesWindow::onDatabaseMessage(sys::Message *msgl)
+    {
+        auto msgNotification = dynamic_cast<db::NotificationMessage *>(msgl);
+        if (msgNotification != nullptr) {
+            if (msgNotification->interface == db::Interface::Name::SMSTemplate) {
+                if (msgNotification->dataModified()) {
+                    rebuild();
+                    return true;
+                }
+            }
+        }
```

Comment:
Let's add `const` before `auto msgNotification`.
Did you consider checking each condition in one `if()` statement?

Reply:
Or if you're not a fan of such chained `if (condition1 && condition2 && ...)` consider rewriting it with early return pattern:

```
if (msgNotification == nullptr) {
    return false;
}

if (msgNotification->interface != db::Interface::Name::SMSTemplate) {
    return false;
}

etc...
```                

Such heavily nested code is much harder to read IMHO.

Reply:
I am not a fan of such chained conditions but in this case, I think it looks just better than the early returns pattern :)